src/main/java/net/minecraftforge/items/pickup/ItemPickupReasons.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.items.pickup;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.world.Container;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.vehicle.MinecartHopper;
+import net.minecraft.world.level.block.entity.HopperBlockEntity;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * Class containing all standard ItemPickupReasons provided by Forge.
+ * <p>Make use of {@link #getOrCreate(String)} for building new {@link ItemPickupReason} instances.
+ */
+@ApiStatus.NonExtendable
+public final class ItemPickupReasons
+{
+    /**
+     * Player is attempting to pick up an item
+     * <p>{@code collector} types should result in the {@link Player} attempting to pick up the item.
+     */
+    public static final ItemPickupReason PLAYER = getOrCreate("player");
+
+    /**
+     * LivingEntity is attempting to pick up an item
+     * <p>{@code collector} types should result in the {@link LivingEntity} attempting to pick up the item.
+     */
+    public static final ItemPickupReason LIVING_ENTITY = getOrCreate("living_entity");
+
+    /**
+     * Hopper is attempting to suck in an item
+     * <p>{@code collector} types should result in the {@link HopperBlockEntity} or {@link MinecartHopper} {@link Container containers} attempting to pick up the item.
+     */
+    public static final ItemPickupReason HOPPER = getOrCreate("hopper");
+
+    /**
+     * {@return all registered pickup reasons}
+     */
+    public static Collection<ItemPickupReason> all()
+    {
+        return Collections.unmodifiableCollection(ItemPickupReason.reasons.values());
+    }
+
+    /**
+     * Gets a given pickup reason or creates a new instance of none exists.
+     * <p>For custom modded reasons, it is recommended you prefix the name with your mod id
+     * <i>[{@code {modId}:{reasonName}}]</i>, this is to prevent registering custom reasons
+     * with duplicate names, and potentially using custom mod reasons in places you did not intend on using them.
```

Comment:
If we're telling people to prefix, then just accept a ResourceLocation

Reply:
to be honest, I personally don't like methods that take in a ResourceLocation, I would rather just pass in a String, it looks cleaner to me than creating a ResourceLocation.

`ItemPickupREasons.getOrCreate("player");`

vs

`ItemPickupReasons.getOrCreate(new ResourceLocation("forge:player"));`
`ItemPickupReasons.getOrCreate(new ResourceLocation("forge", "player"));`

could change the method to take in 2 Strings, `mod id` & `name` which would combine the 2 into a prefixed String or wrap directly into a ResourceLocation instance.

`ItemPickupReasons.getOrCreate("forge", "player");`